library/src/main/java/io/voucherify/android/client/module/Redemption.java

```diff
@@ -0,0 +1,80 @@
+package io.voucherify.android.client.module;
+
+import java.io.IOException;
+
+import io.reactivex.Observable;
+import io.reactivex.Scheduler;
+import io.voucherify.android.client.api.VoucherifyApi;
+import io.voucherify.android.client.callback.VoucherifyCallback;
+import io.voucherify.android.client.exception.VoucherifyError;
+import io.voucherify.android.client.model.VoucherRedemptionContext;
+import io.voucherify.android.client.model.VoucherRedemptionResult;
+import io.voucherify.android.client.utils.RxUtils;
+
+public class Redemption extends AbsModule<Redemption.ExtAsync, Redemption.ExtRxJava>{
+
+    private final VoucherifyApi api;
+
+    private final Scheduler scheduler;
+
+    private final String trackingId;
+
+    Redemption(VoucherifyApi api, Scheduler scheduler, String trackingId) {
+        super(api, scheduler);
+        this.api = api;
+        this.scheduler = scheduler;
+        this.trackingId = trackingId;
+    }
+
+    @Override
+    ExtAsync createAsyncExtension() {
+        return new ExtAsync();
+    }
+
+    @Override
+    ExtRxJava createRxJavaExtension() {
+        return new ExtRxJava();
+    }
+
+    public VoucherRedemptionResult redeem(String code) throws IOException {
+        return api.redeemVoucher(code, trackingId).execute().body();
+    }
+
+    public VoucherRedemptionResult redeem(String code, VoucherRedemptionContext redemptionContext) throws IOException {
```

Comment:
Does backend have two endpoints to redeem a voucher? If no, maybe we can merge our two methods to redeem as they suggest that backend has two endpoints for this.